test/python/primitives/test_sampler.py

```diff
@@ -407,6 +407,13 @@ def test_deprecated_circuit_indices(self, indices):
                 )
             self._compare_probs(result.quasi_dists, target)
 
+    def test_with_shots_option(self):
+        """test with shots option."""
+        params, target = self._generate_params_target([1])
+        with Sampler(circuits=self._pqc) as sampler:
+            result = sampler(circuits=[0], parameter_values=params, shots=1024, seed_primitive=15)
+            self._compare_probs(result.quasi_dists, target)
+
```

Comment:
same as https://github.com/Qiskit/qiskit-terra/pull/8137#discussion_r903752584

Reply:
Added in 1593f34